Sources/PackageCollections/Storage/SQLitePackageCollectionsStorage.swift

```diff
@@ -286,29 +286,38 @@ final class SQLitePackageCollectionsStorage: PackageCollectionsStorage, Closable
     func searchPackages(identifiers: [Model.CollectionIdentifier]? = nil,
                         query: String,
                         callback: @escaping (Result<Model.PackageSearchResult, Error>) -> Void) {
-        self.list(identifiers: identifiers) { result in
-            switch result {
-            case .failure(let error):
-                callback(.failure(error))
-            case .success(let collections):
-                if self.useSearchIndices.get() ?? false {
-                    var matches = [(collection: Model.CollectionIdentifier, package: PackageIdentity)]()
-                    do {
-                        let packageQuery = "SELECT collection_id_blob_base64, repository_url FROM \(Self.packagesFTSName) WHERE \(Self.packagesFTSName) MATCH ?;"
-                        try self.executeStatement(packageQuery) { statement in
-                            try statement.bind([.string(query)])
+        if self.useSearchIndices.get() ?? false {
```

Comment:
The diff is hard to read, but the changes in each method follow the same pattern: if `self.useSearchIndices == true`, call `self.list(identifiers: identifiers)` to fetch collections _after_ we perform the search. This allows us to only fetch the collections we absolutely need.